src/core/qgis.h

```diff
@@ -357,6 +357,20 @@ class CORE_EXPORT Qgis
     };
     Q_ENUM( UnplacedLabelVisibility )
 
+    /**
+     * Flags which control how data providers will scan for sublayers in a dataset.
+     *
+     * \since QGIS 3.22
+     */
+    enum class SublayerQueryFlag : int
+    {
+      FastScan = 1 << 0, //!< Indicates that the provider must scan for sublayers using the fastest possible approach -- e.g. by first checking that a uri has an extension which is known to be readable by the provider
+      ResolveGeometryType = 1 << 1, //!< Attempt to resolve the geometry type for vector sublayers
```

Comment:
Just wondering if a similar flag to return raster data type and bands could be useful in the future.

Reply:
Sure -- I'll add that if it proves to be expensive to calculate